demos/shapes-custom-question-authoring-demo/src/question/index.js

```diff
@@ -0,0 +1,339 @@
+                        if(shapeButton.getAttribute('data-index') === init.response) {
+                            shapeButton.classList.add('selected');
+                        }
+                    })
+                }
+                init.getFacade().disable();
+            }
+
+            
+
+            init.events.trigger('ready');
+        });
+    }
+
+    render() {
+        const { el, init, lrnUtils } = this;
+        const { question, response } = init;
+        
+        // TODO: Requires implementation
+        el.innerHTML = `
+            <div class="${PREFIX} lrn-response-validation-wrapper">
+                <div class="lrn_response_input">
+                    <div class="shapes flex-row"></div>
+                </div>            
+                <div class="${PREFIX}-checkAnswer-wrapper"></div>
+                <div class="${PREFIX}-suggestedAnswers-wrapper"></div>
+            </div>
+        `;
+
+        // Optional - Render optional Learnosity components like Check Answer Button, Suggested Answers List
+        // first before rendering your question's components
+        return Promise.all([
+            lrnUtils.renderComponent('SuggestedAnswersList', el.querySelector(`.${PREFIX}-suggestedAnswers-wrapper`)),
+            lrnUtils.renderComponent('CheckAnswerButton', el.querySelector(`.${PREFIX}-checkAnswer-wrapper`))
+        ]).then(([suggestedAnswersList]) => {
+            this.suggestedAnswersList = suggestedAnswersList;
+            
+            question.shapes && question.shapes.forEach((shape, index) => {
+                const { kind, size, color } = shape;
+                const shapeWapper = document.createElement('div');
+                shapeWapper.setAttribute('class', 'shape-wrapper flex-column');
+                const shapeBtn = document.createElement('button');
+                shapeBtn.setAttribute('class', `shape ${kind}`);
+                shapeBtn.setAttribute('data-index', index)
+                const pxValue = mapSizeStringToFactor(size)*100;
+                if(kind === "octagon") {
+                    shapeBtn.setAttribute('style', `width:${pxValue}px;background:${color};`)
+                }
+                if(kind === "triangle") {
+                    shapeBtn.setAttribute('style', `width:${pxValue+10}px;background:${color};`)
+                }
+                if(kind === "square" || kind === "circle") {
+                    shapeBtn.setAttribute('style', `width: ${pxValue}px; height: ${pxValue}px; background: ${color};`)
+                }
+                if(kind === "hexagon") {
+                    shapeBtn.setAttribute('style', `height: ${pxValue}px; background: ${color};`)
+                }
+                shapeWapper.appendChild(shapeBtn);
+                if(question.display_shape_names) {
+                    const p = document.createElement('p');
+                    p.innerText = kind;
+                    shapeWapper.appendChild(p)
```

Comment:
- new line after the declaration
- use textContent instead of innerText
- typo in shapeWrapper
```suggestion
                    const p = document.createElement('p');

                    p.textContent = kind;
                    shapeWrapper.appendChild(p)
```